app/models/concerns/sluggable.rb

```diff
@@ -10,7 +10,7 @@ module Sluggable
 
     after_initialize :generate_slug, if: -> { slug.blank? }
 
-    validates :slug, presence: true, uniqueness: { case_sensitive: false }
+    validates :slug, presence: true
```

Comment:
this adds a lot of individual lookups when we're creating degrees, and the field is already marked as unique in the database so possibly this can be dropped...